src/painter.rs

```diff
@@ -262,16 +250,23 @@ impl Painter {
         Ok(())
     }
 
-    pub(crate) fn terminal_rows(&self) -> u16 {
+    pub(crate) fn screen_height(&self) -> u16 {
         self.terminal_size.1
     }
 
-    pub(crate) fn terminal_cols(&self) -> u16 {
+    pub(crate) fn screen_width(&self) -> u16 {
         self.terminal_size.0
     }
 
-    pub fn remaining_lines(&self) -> u16 {
-        self.terminal_size.1 - self.prompt_coords.prompt_start.1
+    fn remaining_lines(&self) -> u16 {
+        self.screen_height() - self.prompt_start_row
```

Comment:
Im wondering if only for sanity this should be a saturated sub. I dont know what behaviour we get during screen resizing

Reply:
Good point, but should be safe as resize touches both at the same time. This wasn't the case for the initialization step, next commit unifies that